tests/e2e-tests/specs/front-end/front-end-checkout.test.js

```diff
@@ -0,0 +1,188 @@
+			config.get( 'addresses.customer.billing.lastname' ),
+			config.get( 'addresses.customer.billing.company' ),
+			config.get( 'addresses.customer.billing.country' ),
+			config.get( 'addresses.customer.billing.addressfirstline' ),
+			config.get( 'addresses.customer.billing.addresssecondline' ),
+			config.get( 'addresses.customer.billing.city' ),
+			config.get( 'addresses.customer.billing.state' ),
+			config.get( 'addresses.customer.billing.postcode' ),
+			config.get( 'addresses.customer.billing.phone' ),
+			config.get( 'addresses.customer.billing.email' )
+		);
+	} );
+
+	it( 'Allows customer to fill shipping details', async () => {
+		await CustomerFlow.goToShop();
+		await CustomerFlow.addToCartFromShopPage( 'Simple product' );
+		await CustomerFlow.goToCheckout();
+		await CustomerFlow.productIsInCheckout( 'Simple product', 4, 39.96, 39.96 );
+
+		await expect( page ).toClick( '#ship-to-different-address-checkbox' );
+		await uiUnblocked();
+
+		await CustomerFlow.fillShippingDetails(
+			config.get( 'addresses.customer.shipping.firstname' ),
+			config.get( 'addresses.customer.shipping.lastname' ),
+			config.get( 'addresses.customer.shipping.company' ),
+			config.get( 'addresses.customer.shipping.country' ),
+			config.get( 'addresses.customer.shipping.addressfirstline' ),
+			config.get( 'addresses.customer.shipping.addresssecondline' ),
+			config.get( 'addresses.customer.shipping.city' ),
+			config.get( 'addresses.customer.shipping.state' ),
+			config.get( 'addresses.customer.shipping.postcode' )
+		);
+	} );
+
+	it( 'Allows guest customer to place order', async () => {
+		await CustomerFlow.goToShop();
+		await CustomerFlow.addToCartFromShopPage( 'Simple product' );
+		await CustomerFlow.goToCheckout();
+		await CustomerFlow.productIsInCheckout( 'Simple product', 5, 49.95, 49.95 );
+		await CustomerFlow.fillBillingDetails(
+			config.get( 'addresses.customer.billing.firstname' ),
+			config.get( 'addresses.customer.billing.lastname' ),
+			config.get( 'addresses.customer.billing.company' ),
+			config.get( 'addresses.customer.billing.country' ),
+			config.get( 'addresses.customer.billing.addressfirstline' ),
+			config.get( 'addresses.customer.billing.addresssecondline' ),
+			config.get( 'addresses.customer.billing.city' ),
+			config.get( 'addresses.customer.billing.state' ),
+			config.get( 'addresses.customer.billing.postcode' ),
+			config.get( 'addresses.customer.billing.phone' ),
+			config.get( 'addresses.customer.billing.email' )
+		);
+		await uiUnblocked();
+
+		await expect( page ).toClick( '.wc_payment_method label', { text: 'Cash on delivery' } );
+		await expect( page ).toMatchElement( '.payment_method_cod', { text: 'Pay with cash upon delivery.' } );
+		await uiUnblocked();
+		await CustomerFlow.placeOrder();
+
+		await expect( page ).toMatch( 'Order received' );
+
+		// Get order ID from the Thank you page URL
+		let reg = /.+?\:\/\/.+?(\/.+?)(?:#|\?|$)/;
```

Comment:
I agree with you that getting order ID from HTML is more stable than from URL since it can change in the future if we decide to set different permalinks settings. Thanks for pointing it out. Fixed it as you suggested.